src/js/components/DataTable/Body.js

```diff
@@ -179,22 +180,45 @@ const Body = forwardRef(
     const [active, setActive] = React.useState();
     const [lastActive, setLastActive] = React.useState();
 
+    const selectCheckBox = () => {
+      const primaryValue = data[active]?.[primaryProperty];
+      if (selected && selected.includes(primaryValue)) {
+        onSelect(selected.filter((s) => s !== primaryValue));
+      } else onSelect([...selected, primaryValue]);
+    };
+
     return (
       <Keyboard
-        onEnter={
-          // active is undefined if user never used the keyboard
-          onClickRow &&
-          active >= 0 &&
-          (!disabled ||
-            !disabled.includes(datumValue(data[active], primaryProperty)))
-            ? (event) => {
-                event.persist();
-                const adjustedEvent = event;
-                adjustedEvent.datum = data[active];
-                onClickRow(adjustedEvent);
-              }
-            : undefined
-        }
+        onEnter={(event) => {
+          if (
+            onClickRow &&
```

Comment:
Could we potentially pull this out into a variable and place before the return:

```
const clickedRow = onClickRow && active >= 0 &&
            (!disabled ||
              !disabled.includes(datumValue(data[active], primaryProperty)))
```

And then use that in the onEnter and onSpace as 

```
if (clickedRow) {
...
}